src/servers/http_server_v2.cc

```diff
@@ -1110,23 +1110,32 @@ HTTPAPIServerV2::EVBufferToInput(
   //
   // Get the addr and size of from the evbuffer.
   int buffer_len = evbuffer_get_length(input_buffer);
-  char json_buffer[buffer_len];
+
+  std::vector<char> complete_buffer(buffer_len);
```

Comment:
You can copy into complete_buffer.data() below without reserving space in the vector. But copying the entire body into a buffer is something we can't do because of performance. Note how the original code was careful to use the data directly from the evbuffer, even if that meant it had to cross multiple blocks. We need to continue to do the same thing. Even for json parsing we only want to copy if the json part is not in a contiguous buffer.

Reply:
Done